### ISSUE TITLE
incorrect handling of "D" in existing CIGAR strings with mask_* functions

### ISSUE THREAD

**Issue by ahstram** [brentp/cigar#2]
Here's an example:

`print("3M1D137M",cigar.Cigar("3M1D137M").mask_left(4).cigar)`

Which will yield:

> 3M1D137M 5S136M

Which is incorrect, as the original "3M1D137M" string has query sequence length 140, but the left-masked query sequence length of "5S136M" is 141.

**Comment by brentp** [comment]
hi, looks like a bug. I'd gladly accept a pull request to fix this, but I haven't used this library for a while and don't have time to invest in it.

**Comment by ahstram** [comment]
Thanks for the quick response. I'll take a stab at it and submit a PL. There's similar code in the picard code base ("clip3primeEndsTo5primeEnds()", I believe) which seems to consider the "D".

**Comment by ahstram** [comment]
Just submitted PR #3

**Comment by Asutu** [comment]
I also bumped  into this issue while playing with the library. Are you still accepting PRs? If so, I'm happy to submit mine for your review. It seems to be working correctly and also added a few tests.

**Comment by brentp** [comment]
yes, I'll accept PR's with test. I don't use this, so you're certainly welcome to just fork and maintain.
